### ISSUE TITLE
Generate CRD based via controller-gen

### ISSUE THREAD

**Issue by youngbupark** [dapr/dapr#2355]
## In what area(s)?

/area runtime

## Describe the feature

The current code-generate makefile target does not generate crd so that all markers in types.go are not applied to the actual [crd](https://github.com/dapr/dapr/tree/master/charts/dapr/crds).

adding the options to code-generate, it will generate crd based on markers in types.go under https://github.com/dapr/dapr/tree/master/pkg/apis

then we need to add crd output option `output:crd:artifacts:config=config/crd/bases`

https://github.com/dapr/dapr/blob/28a15366c761b42325a576c98ace3090b0b05240/tools/codegen.mk#L8-L10

e.g.

```
code-generate: controller-gen
	$(CONTROLLER_GEN) object:headerFile="./tools/boilerplate.go.txt" \
		crd:trivialVersions=true paths="./pkg/apis/..." output:crd:artifacts:config=config/crd/bases
```

## Release Note

RELEASE NOTE: **ADD** the option to generate crd via controller-gen

**Comment by youngbupark** [comment]
@yaron2 markers in types.go are not applied to our crd yaml. to use the markers, we need to generate crds via controller.gen as well, not a manual way.

**Comment by mukundansundar** [comment]
Fixed with #2356
